test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -10131,18 +10163,27 @@ def __init__(self, hostname=None, server=None, pbsconf_file=None,
         rg = self.parse_resource_group(self.hostname, self.resource_group_file)
         self.resource_group = rg
 
+        self.sc_name = id
+        self.db_access = db_access
+
+        self.version = None
+
+        self.holidays_obj = Holidays()
+        self.holidays_parse_file(level=logging.DEBUG)
+
+    def initialise_service(self):
+        """
+        initialise the scheduler object
+        """
         try:
             attrs = self.server.status(SCHED, level=logging.DEBUG,
-                                       db_access=db_access)
+                                       db_access=self.db_access,
+                                       id=self.sc_name)
             if attrs is not None and len(attrs) > 0:
                 self.attributes = attrs[0]
         except (PbsManagerError, PbsStatusError), e:
             self.logger.error('Error querying scheduler %s' % e.msg)
-
-        self.version = None
-
-        self.holidays_obj = Holidays()
-        self.holidays_parse_file(level=logging.DEBUG)
+        PBSService.initialise_service(self)
```

Comment:
Please call the base class method i.e. PBSService.initialise_service(self) at the top of this function as we are overriding this function here.